### DISCUSSION TITLE
pylint.rc file

### DISCUSSION THREAD

**Discussion by RaGrCobham** [priv-kweihmann/meta-sca#9176] [General]
I'm confused why the pylint.rc directory is hardcoded to reside in 'T' directory.

In our use case, we don't ship anything from build folder.

**Comment by priv-kweihmann** [comment]
T is the perfect place IMO to put per-workspace files to.
And this pylint.rc file is needed to make pylint accept import paths apart from the standard paths. As you can see here https://github.com/priv-kweihmann/meta-sca/blob/86d62c8da154c4cfb5436a0ede0633b61ecc14f2/classes/sca-pylint-core.bbclass#L80 it's the sole purpose of that file as part of pylint's meta-sca integration.

One could place that file anywhere else inside the workspace, but most of the meta-sca intermediate files and results are stored in T, it makes sense to keep that there as well.
As T won't get cleaned by rm_work class, one can also take a look at all these files, which helps debugging a lot

**Reply by RaGrCobham** [reply]
> One could place that file anywhere else inside the workspace

As long as I change do_sca_pylint_core() to use a separate variable for the path, right?

> T is the perfect place IMO to put per-workspace files to.

But if a per-project situation is desired? I'm not familiar with having build paths inside repositories.

**Reply by priv-kweihmann** [reply]
project is the same as a workspace - workspace like mentioned in the variable WORKDIR - so it's project@version == workspace.
And you need to make the dependencies which are installed to WORKDIR/recipe-sysroot/usr/lib/python*/site-packages/* visible to pylint, as otherwise they won't be seen by the tool
